### PR TITLE
Governance for the Vega Organization

### PR DESCRIPTION
**Pull request by domoritz**
Corresponding updates to the Altair organization https://github.com/altair-viz/altair/pull/3277.

### CODE REVIEW

File: GOVERNANCE.md
Comment (domoritz) [inline GOVERNANCE.md:56]: @arvind, @domoritz, @jheer, @kanitw, @lsh please approve that you want to be an admin

File: GOVERNANCE.md
Comment (domoritz) [inline GOVERNANCE.md:65]: @binste @ChristopherDavisUCI @jakevdp @joelostblom @mattijn @stsievert please approve that you want to be an admin

File: GOVERNANCE.md
Comment (domoritz) [inline GOVERNANCE.md:38]: @jonmmease please approve that you want to be a maintainer

File: GOVERNANCE.md
Comment (domoritz) [inline GOVERNANCE.md:34]: @fandu-db @joelostblom @jonmmease @keckelt @mattijn @yhoonkim please approve that you want to be a maintainer

**Comment by binste** [comment]
Thanks @domoritz for putting this together! I like the descriptions of the different roles. I'm wondering how we can best integrate this with the existing governance structure for `altair-viz` which we merged in https://github.com/altair-viz/altair/pull/3139. How about just linking to the https://github.com/altair-viz/altair/tree/main/governance folder for all things related to Altair? Steering committee members are already listed [here](https://github.com/altair-viz/altair/blob/main/governance/org-docs/STEERING-COMMITTEE.md) and maintainers [here](https://github.com/altair-viz/altair/blob/main/governance/project-docs/MAINTAINERS.md).

**Comment by domoritz** [comment]
Ah, I missed that. Thanks for the pointer. 

I think we have two options. One, leave the governance documents mostly separate and link to the Altair governance from the Vega governance but don't say that the Vega governance governs Altair. Altair could still delete its own CoC and refer to the Vega one to reduce redundancy but otherwise we keep things separate. Two, merge and move the parts we like about governance from Altair into this new governance. I'm open to either option and leave it up to the current Altair steering committee to decide.

**Comment by binste** [comment]
We just had an Altair steering committee meeting yesterday where we also discussed governance and how interactions could look like with the other Vega projects (I'll try to add the minutes soon [here](https://github.com/altair-viz/altair/discussions/3164). Your PR went live during the meeting so we unfortunately did not yet discuss it. :)

A change in the Altair governance should happen based on a committee vote. I'll try to set up the next committee call in ~1 month. If you already want to go ahead and merge, I'd suggest to simply link to https://altair-viz.github.io/about/governance.html for Altair. In that call we could discuss if we want to

* move some of the governance documentation over to this repo
* adapt the same role definitions for maintainers and admins which are slightly different to what we have now

I'd be very curious to hear if there are any plans to set up a similar governance body to the Altair steering committee for Vega/Vega-Lite. In a second step, maybe even incl. Altair? We discussed in our call yesterday that for the next Altair committee meeting, we'll extend invites to the broader Vega maintainer/contributor community in case there is interest to discuss the collaboration and overlap in governance between Vega/VL and Altair.

**Review by jonmmease — APPROVED** [review]
Thanks for writing this up @domoritz, I'm happy with the Vega project governance info, and would agree with linking to Vega-Altair's governance documents for the purpose of this PR, and then the Vega-Altair steering committee can discuss the possibility of moving our steering committee documents upstream to this repo.

One thing we need to decide is whether there is still a need for a Vega-Altair specific steering committee once we have a larger Vega steering committee.  As @binste mentioned, we decided to invite the Vega and Vega-Lite maintainers to our next steering committee meeting so that we can discuss these things together.

Really glad to be having these conversations!

**Comment by domoritz** [comment]
Re the steering committee. I like the idea of an overall Steering Committee but we could have sub-committees. If we do that, we definitely want a single charter document and I'd be happy to just adopt the one from Altair with some modifications.

Maintainers should be separate. I haven't quite figured out a lightweight way to define maintainers for all the various Vega projects (tooltip, embed, etc). My current thinking is that they all share the same maintainers. I would also propose that a project may define separate admins that are allowed to make releases. 

This way we would have four roles/levels.

* Contributor (anyone who makes any contribution, needs to adhere to CoC)
* Maintainer (maintain project, merge PRs etc, appointed by maintainers)
* Admin (maintainer who can also make a release, appointed by existing admins, may only exist if this is a separate role from maintainers)
* Steering committee (Vega Project wide, can have subcommittee if desired)

Maintainers and admins directly correspond to the roles in GitHub.

File: GOVERNANCE.md
Comment (domoritz) [reply GOVERNANCE.md]: Resolved since Altair has their own docs for now.

File: GOVERNANCE.md
Comment (domoritz) [reply GOVERNANCE.md]: Resolved since Altair has their own docs for now.

**Comment by mattijn** [comment]
Great to see that this is being pushed forward @domoritz! I like the idea of sub-committees with a single charter document. One question, within the Vega organization there are more repositories than currently mentioned in this governance model. Could you reflect a bit on this? Most notably, vl-convert, which is crucial for Altair nowadays.

**Review by joelostblom — APPROVED** [review]
Thanks for putting this together and for everyone's helpful comments! I think this looks good overall and I'm ok either merging and updating in another PR once we have had the call everyone together or keeping this PR open until we have had that call and made the updates.

I think the four levels of roles sounds good; "contributor" requires no management from our side, "maintainer/admin" is something we already have in the GitHub roles and it is great that there is now a formal description of these roles, "steering committee" is really the only new role and I think it will be an important one to coordinate things across the vega ecosystem so I'm excited to see that added here.

**Comment by domoritz** [comment]
> One question, within the Vega organization there are more repositories than currently mentioned in this governance model. Could you reflect a bit on this? Most notably, vl-convert, which is crucial for Altair nowadays.

I forgot to include Vega-Lite-convert. I'll add it.

Yes, there are more but many are not active (voyager, compassql) or have very specific purposes (typescript-json-schema). One thing I like about the charter is that it explains how to add projects, which we will want for Vega.

File: GOVERNANCE.md
Comment (kanitw) [inline GOVERNANCE.md:23]: > via Slack reactions 

Is this slack reaction private within the maintainer channel?  (Otherwise, I doubt if anyone will feel comfortable "thumbing down" in public.)

File: GOVERNANCE.md
Comment (kanitw) [inline GOVERNANCE.md:41]: I think we need some RFC template so people know to enumerate alternative design etc. 
Github issue is generally quite ineffective to enumerate such options.

File: GOVERNANCE.md
Comment (domoritz) [reply GOVERNANCE.md:23]: Yes, it's in the private channel.

File: GOVERNANCE.md
Comment (domoritz) [reply GOVERNANCE.md:41]: Slack is even worse for it and doesn't even have templates. And yes, I agree having some template would be useful but it should be lightweight. I honestly think our pull request templates are not as useful as we had hoped (people don't fill it out and it clutters the pull request description).

File: GOVERNANCE.md
Comment (jheer) [inline GOVERNANCE.md:23]: "lower the barrier the enter" -> "lower the barrier to entry"

**Review by jheer — APPROVED** [review]

File: GOVERNANCE.md
Comment (arvind) [inline GOVERNANCE.md:23]:
```suggestion
To become a Maintainer, a contributor must be nominated by an existing Maintainer and approved by a majority of Maintainers via Slack reactions within 7 days (simple majority of responses). A new Maintainer should be added to 1) the GitHub team, 2) the Slack channel, and 3) the [MAINTAINERS.md file](MAINTAINERS.md). We want more people to become Maintainers and lower the barrier the enter. If you are interested in becoming a Maintainer, please reach out to an existing Maintainer.
```

File: GOVERNANCE.md
Comment (arvind) [inline GOVERNANCE.md:29]: Not sure what "who can make" means? I don't think we have a Makefile anymore?

File: GOVERNANCE.md
Comment (arvind) [reply GOVERNANCE.md:41]: I agree that our PR template hasn't worked well because it's heavyweight (lots of checkboxes). I think one other reason is because the wording/structure makes it a little unclear how to actually fill things out (e.g., reads like instructions, with links to advice that need to be deleted; some checkboxes, some bullet points; and, unclear where one should add a description of the PR, etc.). 

So, we could use something more lightweight like:

```
## PR Description

## Checklist

- [ ] This PR is atomic (i.e., it fixes one issue at a time).
- [ ] The title is a concise [semantic commit message](https://www.conventionalcommits.org/) (e.g. "fix: correctly handle undefined properties").
- [ ] `yarn test` runs successfully
... etc
```

**Review by arvind — APPROVED** [review]
Thanks for doing this @domoritz, LGTM!

File: GOVERNANCE.md
Comment (domoritz) [reply GOVERNANCE.md:29]: I meant to say "who can release new versions and are also responsible for doing so". I think in general we should say that any power also comes with the corresponding responsibility. So if you can release new versions, you are also expected to make releases when necessary. 

"Make a release" === "release a new version". I didn't mean to imply any relationship to Make. Can you edit the text to make it clearer?

**Review by kanitw — APPROVED** [review]

File: GOVERNANCE.md
Comment (domoritz) [reply GOVERNANCE.md]: Done in https://github.com/vega/vega-lite/pull/9187

**Comment by domoritz** [comment]
I made some updates based on the discussion today (link to meeting minutes coming from @binste). 

I did remove the antitrust and trademark documents since I don't think we need them right now.

**Review by jonmmease — APPROVED** [review]
Thanks for pulling this together @domoritz, looks great

File: project-docs/ADMINS.md
Comment (ChristopherDavisUCI) [inline project-docs/ADMINS.md:17]: I think we've been using a hyphen in the name, so Vega-Altair.

**Review by ChristopherDavisUCI — APPROVED** [review]
Thank you for assembling this so quickly!  I just left one minor comment.

File: project-docs/ADMINS.md
Comment (domoritz) [reply project-docs/ADMINS.md:17]: Done

File: CHARTER.md
Comment (jheer) [inline CHARTER.md:17]: Typo? “Project approval and oversight” Should it be “project approval oversight”?

**Review by jheer — APPROVED** [review]

File: CHARTER.md
Comment (binste) [inline CHARTER.md:5]:
```suggestion
This is the organizational charter for [Vega](https://github.com/vega/vega), [Vega-Lite](https://github.com/vega/vega-lite), [Vega-Altair](https://github.com/altair-viz/altair), [Vega-Embed](https://github.com/vega/vega-embed), [Vega-Datasets](https://github.com/vega/vega-datasets), [Vega-Themes](https://github.com/vega/vega-themes), [Vega-Tooltips](https://github.com/vega/vega-tooltip), [Vega-Editor](https://github.com/vega/editor), [VL-Convert](https://github.com/vega/vl-convert), [Altair-Tiles](https://github.com/altair-viz/altair_tiles), and [dash-vega-components](https://github.com/altair-viz/dash-vega-components). These projects are collectively referred to as the "Vega projects" in this document.
```
Added altair_tiles and dash-vega-components as projects as they are also under the current Vega-Altair governance. I'll create separate PRs in these projects to remove the `project-docs` and just refer to this repo.

**Review by binste — COMMENTED** [review]
Thanks again @domoritz ! Only added one small comment.

**Review by binste — APPROVED** [review]

File: CHARTER.md
Comment (domoritz) [reply CHARTER.md:5]: Good points. I added a note below about adding projects to the umbrella. Alternatively, we can just say all projects under the two github organizations are under the charter. Any reason we shouldn't do that?

File: STEERING-COMMITTEE.md
Comment (domoritz) [reply STEERING-COMMITTEE.md:9]: Done

File: CHARTER.md
Comment (mattijn) [reply CHARTER.md]: From my thinking it make sense to include all projects within the repo. 
One of the reason why I think it make sense is because there a few packages within the Vega-Altair repo that can potentially be deprecated/archived, but I think this requires a vote by the maintainers of the Altair-viz organization. 
Same is probably valid for some of the projects within the Vega organization I think.

File: CHARTER.md
Comment (domoritz) [reply CHARTER.md]: Updated to allow 1 or 2 chairs.

File: CHARTER.md
Comment (domoritz) [reply CHARTER.md:17]: I copied this from Altair. I think it means approving projects and overseeing projects. The steering committee is responsible for approving projects and not just overseeing the approval. Makes sense? We could shorten this to "project approval" since we have a separate point for "policy oversight". I'll change it for now.

**Review by joelostblom — APPROVED** [review]
Looks great, thank you for putting it together!

File: CHARTER.md
Comment (arvind) [inline CHARTER.md:39]:
```suggestion
* **Unreachable Member**. If a member is unresponsive at their listed handle for more than three months the Steering Committee may vote to remove the member.
```

File: STEERING-COMMITTEE.md
Comment (arvind) [inline STEERING-COMMITTEE.md:11]:
```suggestion
| Arvind Satyanarayan | @arvind | MIT CSAIL |
```

File: project-docs/ADMINS.md
Comment (arvind) [inline project-docs/ADMINS.md:11]:
```suggestion
| Arvind Satyanarayan | @arvind | MIT CSAIL |
```

File: project-docs/GOVERNANCE.md
Comment (arvind) [inline project-docs/GOVERNANCE.md:21]: Just want to double check a couple of things here:

* Do these two lines mean that we'll be switching the `-dev` channels from public to private on the Vega slack?
* Will votes for maintainers (and actually any membership level) occur on these private channels? If so, there is some potential for awkward power dynamics—eg person A is nominated; there is an up/down vote, they're ultimately added as a Maintainer and join the private channel and can now see who voted against their membership. I imagine this will be a relatively rare event, so I don't think we need to overengineer it. I'm also on the fence about how much of an issue this actually is. But, one lightweight approach might be for whomever is chairing the nomination process to announce the result at the conclusion of the vote (eg as number for/against), and then delete the Slack post corresponding to the vote.

File: project-docs/MAINTAINERS.md
Comment (arvind) [inline project-docs/MAINTAINERS.md:11]:
```suggestion
| Arvind Satyanarayan | @arvind | MIT CSAIL |
```

**Review by arvind — APPROVED** [review]
This looks really fantastic!! HUGE thank you @domoritz for driving this, and to everyone else for contributing. 

I just updated my affiliations throughout and left one comment about potential dynamics issues of voting on people's roles as part of Slack channels that they may subsequently join. Not sure it needs to be addressed as part of the governance documents, but just wanted to make sure it was on our radar so it doesn't catch us unawares the first few times we need to run the process through.

**Review by lsh — APPROVED** [review]

File: project-docs/GOVERNANCE.md
Comment (domoritz) [reply project-docs/GOVERNANCE.md:21]: For channels, I made new ones just to now kick people out of channels but we can basically "archive" the existing ones at some point. 

Hm, right, I hadn't thought about the awkwardness of being able to see who voted against you. @jheer suggested in the steering committee meeting that we could use Google forms for anonymous voting. We could switch to that for votes on people. I would suggest that we restrict access to the results to the chairs and they should delete the form after the vote has passed similar to what you suggest for Slack votes. 

I'd say for now let's do what you suggested and delete the vote post after the vote.

**Review by yhoonkim — APPROVED** [review]

**Review by fandu-db — APPROVED** [review]

**Review by mattijn — APPROVED** [review]

**Review by keckelt — APPROVED** [review]
Looks great, thank you!

**Comment by domoritz** [comment]
Waiting for last review from @kanitw before merging.

**Review by kanitw — APPROVED** [review]

**Pull request merged**
